src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -553,7 +553,7 @@ class Address {
           i->sf(offset(), 20, 12);
         } else {
           i->f(0b01, 25, 24);
-          i->f(offset() >> size, 21, 10);
+          i->f(checked_cast<unsigned>(offset() >> size), 21, 10);
```

Comment:
I remember there being issues with checked_cast and sign extension.  When going from int64_t to unsigned and back, I think we need to do int64_t --> int32_t --> unsigned, and not int64_t --> uint64_t --> unsigned.  Is that what checked_cast will do?  To be safe, or at least make it easier to understand, shound't we use checked_cast only to change the size or sign, but not both?  So going to int64_t to unsigned would require two checked_casts.

Reply:
That's right.